graph_1.py

```diff
@@ -0,0 +1,61 @@
+class Graph:
+    def __init__(self, iterable=None):
+        self.graph = {}
+        if iterable is not None:
+            for item in iterable:
+                self.add_node(item)
+
+    def add_node(self, node):
+        if node not in self.graph:
+            self.graph[node] = []
```

Comment:
Is there an advantage of a list here over a set()?  What about vise versa?

Reply:
The list in this situations only advantage is insertion order preservation, however a set would have the benefit of if someone added the same edge twice, it would not record 2 edges. There is probably more, but these are all that come to mind.

Reply:
Yup, typically a set would be used here for the advantages you have said, as well as the look up speed.  Anywhere you use 'edge in graph@node' is currently O(n) but with a set this becomes O(1).

Can you switch it to be a set?